_assets/js/templates/paginationTemplate.js

```diff
@@ -0,0 +1,80 @@
+// Imports:
+//= require ../utils/getOffsetParam.js
+//= require ../utils/setOffsetParam.js
+//= require ./nextButtonTemplate.js
+//= require ./previousButtonTemplate.js
+//= require ./ellipsisTemplate.js
+//= require ./paginationButtonTemplate.js
+
+const paginationTemplate = (resultsArr, urlParams) => {
+  let offsetInt = parseInt(urlParams.get("offset"));
+  if (resultsArr.length <= 5) {
+    return `
+        ${previousButton()}
+        ${resultsArr
+          .map(function (offsetValue, index) {
+            paginationButton(offsetValue, index);
+          })
+          .join(" ")}
+        ${nextButton()}`;
+  } else {
+    return `
+        ${previousButton()}
+        ${paginationButton(resultsArr[0], 0)}
+        ${offsetInt === 0 || offsetInt === 20 || offsetInt === 40 ? '' : ellipsis()}
+        ${resultsArr
+          .map(function (offsetValue, index) {
+            // If it is either the first page or last page, render nothing. The first and last page buttons are fixed.
+            if (
+              index === 0 ||
+              index === resultsArr.indexOf(resultsArr[resultsArr.length - 1])
+            ) {
+              return;
+              // If you are on the first page, render the next three page buttons.
+            } else if (
+              offsetInt === 0 &&
+              (offsetValue === offsetInt + 20 ||
+                offsetValue === offsetInt + 40 ||
+                offsetValue === offsetInt + 60)
```

Comment:
consider making this a named function that returns a boolean for clarity.  

for example 
```
else if (pageTwoThreeOrFour(offsetInt, offsetValue))

. . . 

function pageTwoThreeOrFour(offsetInt, offsetValue) { 
 return   offsetInt === 0 &&
              (offsetValue === offsetInt + 20 ||
                offsetValue === offsetInt + 40 ||
                offsetValue === offsetInt + 60)
}

```

Reply:
Donezo. See new commit.